controlplane/modules/route/internal/rib/map_trie.go

```diff
@@ -0,0 +1,70 @@
+package rib
+
+import (
+	"net/netip"
+)
+
+type MapTrieKey struct {
+	Prefix netip.Prefix
+}
+
+type MapTrie [129]map[MapTrieKey]*RoutesList
+
+func NewMapTrie(preAllocSize int) MapTrie {
+	mt := MapTrie{}
+	for idx := range mt {
+		mt[idx] = make(map[MapTrieKey]*RoutesList, preAllocSize)
+	}
+	return mt
+}
```

Comment:
Suggest better naming - `capacity`.

```suggestion
func NewMapTrie(cap int) MapTrie {
	mt := MapTrie{}
	for idx := range mt {
		mt[idx] = make(map[MapTrieKey]*RoutesList, cap)
	}
	return mt
}
```